packages/react-network/src/test/hooks.test.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import {mount} from '@shopify/react-testing';
+import {Header} from '@shopify/network';
+import {NetworkManager} from '../manager';
+import {NetworkContext} from '../context';
+import {useAcceptLanguages} from '../hooks';
+
+describe('useAcceptLanguages()', () => {
+  function MockComponent({fallback}: {fallback?: string}) {
+    const [local] = useAcceptLanguages(fallback);
+
+    return <>{local}</>;
+  }
+
+  it('returns the locale from the language header', async () => {
+    const lang = 'it';
+    const wrapper = await mountWithLanguageHeader(<MockComponent />, lang);
+
+    expect(wrapper.find(MockComponent)).toContainReactText(lang);
+  });
+
+  it('returns the first locale from the language header with multiple languages', async () => {
+    const lang = 'it, fr, en-US';
```

Comment:
IIRC the actual header also has percentage matches that we'll need to strip out of the string (example from MDN: `fr-CH, fr;q=0.9, en;q=0.8, de;q=0.7, *;q=0.5`)

Reply:
Yes you are correct. I am going to install a module to do the actual parsing here.